lib/appmap/handler/net_http.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+require 'appmap/event'
+
+module AppMap
+  module Handler
+    class HTTPClientRequest < AppMap::Event::MethodEvent
+      attr_accessor :request_method, :url, :params, :headers
+
+      def initialize(http, request)
+        super AppMap::Event.next_id_counter, :call, Thread.current.object_id
+
+        path, query = request.path.split('?')
+        query ||= ''
+
+        protocol = http.use_ssl? ? 'https' : 'http'
+        port = if http.use_ssl? && http.port == 443
+          nil
+        elsif !http.use_ssl? && http.port == 80
+          nil
+        else
+          ":#{http.port}"
+        end
+
+        url = [ protocol, '://', http.address, port, request.path ].compact.join
+
+        self.request_method = request.method
+        self.url = url
+        self.headers = AppMap::Util.select_headers(NetHTTP.request_headers(request))
+        self.params = Rack::Utils.parse_nested_query(query)
+      end
+
+      def to_h
+        super.tap do |h|
+          h[:http_client_request] = {
+            request_method: request_method,
+            url: url,
+            headers: headers
+          }.compact
+
+          unless params.blank?
+            h[:message] = params.keys.map do |key|
+              val = params[key]
+              {
+                name: key,
+                class: val.class.name,
+                value: self.class.display_string(val),
+                object_id: val.__id__,
+              }
+            end
+          end
```

Comment:
This won't have complete URL then, because the query parameters end up in `message`, right? Is it more convenient this way? Should the spec proposal be revised?

BTW, it doesn't make any sense to put `object_id` or `class` there, they're all going to be strings that have just been created. I know the spec requires it, but perhaps we should make them optional.